internal/uvm/update_uvm.go

```diff
@@ -0,0 +1,57 @@
+package uvm
+
+import (
+	"context"
+
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func (uvm *UtilityVM) UpdateConstraints(ctx context.Context, data interface{}, annotations map[string]string) error {
+	var memoryLimitInBytes *uint64
+	var processorLimits *hcsschema.ProcessorLimits
+
+	switch resources := data.(type) {
+	case *specs.WindowsResources:
+		if resources.Memory != nil {
+			memoryLimitInBytes = resources.Memory.Limit
+		}
+		if resources.CPU != nil {
+			processorLimits := &hcsschema.ProcessorLimits{}
+			if resources.CPU.Maximum != nil {
+				processorLimits.Limit = uint64(*resources.CPU.Maximum)
+			}
+			if resources.CPU.Shares != nil {
+				processorLimits.Weight = uint64(*resources.CPU.Shares)
+			}
+		}
+	case *specs.LinuxResources:
+		if resources.Memory != nil {
+			mem := uint64(*resources.Memory.Limit)
+			memoryLimitInBytes = &mem
```

Comment:
I don't understand this. In case of windows resources above we directly assign the value but there we assign it to this `mem` variable and then assign it to `memoryLimitInBytes` why?

Reply:
I also don't see why the `memoryLimitInBytes` is a pointer. It can just be a `uint64` right?

Reply:
I did it like this since the value of resources.Memory.Limit is not a uint64 pointer, so I needed to convert it. It's a pointer just so that later in the function we can determine if the user actually set a memory limit to update versus just a default value of 0. Does that make sense?

Reply:
LinuxResources specifically uses `int64` for memory limit so that one can specify memory limit as `-1` to say no memory limit (https://github.com/opencontainers/runtime-spec/commit/e73cd70caac6ec35ff8153847491352236632417). But I guess even if we treat that -1 (int64) as a uint64 we are practically going to get unlimited memory so it should be okay.
Using pointer is fine.

Reply:
I could do a specific check that the value isn't negative?

Reply:
I don't think that's necessary but you can if you want to.